### PR TITLE
Change logging of new connections from warn to info level

### PR DESCRIPTION
**Pull request by ishantanu**
Signed-off-by: Shantanu Deshpande <shantanud106@gmail.com>

**Summary of changes**:
1. Changed the log level for new connections to `debug` level instead of `warn`.

Fixes: #3557

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by aanm** [comment]
@iamShantanu101 thanks for the PR, I think those types of messages should be `debug` cc @rlenglet

**Comment by aanm** [comment]
test-me-please

**Comment by ishantanu** [comment]
Can I get a confirmation if this needs to be changed to `debug`? If yes, I'll make the changes and rebase the branch against master as well. cc @aanm @rlenglet

**Comment by rlenglet** [comment]
@iamShantanu101 Yes please change this to debug. Thanks!

File: envoy/cilium_network_filter.cc [hunk 1]
```diff
@@ -52,7 +52,7 @@ namespace Filter {
 namespace CiliumL3 {
 
 Network::FilterStatus Instance::onNewConnection() {
-  ENVOY_LOG(warn, "Cilium Network: onNewConnection");
+  ENVOY_LOG(info, "Cilium Network: onNewConnection");
```
Comment (rlenglet) [inline envoy/cilium_network_filter.cc]: As @aanm commented, this should be `debug`. Thanks for fixing this!

**Review by rlenglet — CHANGES_REQUESTED** [review]

**Comment by rlenglet** [comment]
CI build failure is unrelated to this PR (https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/2184/).

**Comment by rlenglet** [comment]
test-me-please

**Comment by ishantanu** [comment]
Updated to `debug` @rlenglet

**Comment by rlenglet** [comment]
test-me-please

**Review by rlenglet — APPROVED** [review]

**Pull request merged**